metals/src/main/scala/scala/meta/internal/tvp/ClasspathTreeView.scala

```diff
@@ -0,0 +1,146 @@
+package scala.meta.internal.tvp
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Symbol
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.SymbolInformation.{Kind => k}
+import scala.meta.internal.mtags.GlobalSymbolIndex
+
+/**
+ * A tree view that exposes a package explorer to browse symbols such as classes, traits and methods.
+ *
+ * @param A The metadata used to render the toplevel nodes such as jar files or
+ *          a Scala build target (with scala version, classpath, ...).
+ * @param B The identifier key that is persisted into the URI of tree view nodes,
+ *          such as a jar file URI or a build target URI.
+ */
+class ClasspathTreeView[A, B](
```

Comment:
> [A, B]

the comments are quite detailed, what do you think about some more detailed names, so the developer doesn't have to scroll up here?